engines/osxnative/engine.go

```diff
@@ -0,0 +1,51 @@
+//go:generate go-composite-schema --unexported --required engine payload-schema.yml generated_payloadschema.go
+
+// Package osxnative implements the Mac OSX engine
+package osxnative
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/engines/extpoints"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type engine struct {
+	engines.EngineBase
+	log *logrus.Entry
+}
+
+type engineProvider struct {
+	extpoints.EngineProviderBase
+}
+
+func init() {
+	// Register the mac engine as an import side-effect
+	extpoints.EngineProviders.Register(engineProvider{}, "macosx")
+}
+
+func (e engineProvider) NewEngine(options extpoints.EngineOptions) (engines.Engine, error) {
+	return engine{log: options.Log}, nil
+}
+
+func (e engineProvider) ConfigSchema() runtime.CompositeSchema {
+	return runtime.NewEmptyCompositeSchema()
+}
+
+func (e engine) PayloadSchema() runtime.CompositeSchema {
+	return payloadSchema
+}
+
+func (e engine) NewSandboxBuilder(options engines.SandboxOptions) (engines.SandboxBuilder, error) {
+	taskPayload := options.Payload.(*payload)
+	return sandboxbuilder{
+		engines.SandboxBuilderBase{},
```

Comment:
I'm not sure how you feel, and this is totally a nit on my part, but how about naming each of these fields?  It makes it explicit what fields are being set to what values and doesn't rely on ordering to assign them.